src/main/java/frc/robot/VisionProcessor.java

```diff
@@ -132,19 +145,34 @@ public void process(Mat sourceFrame) {
         ArrayList<RotatedRect> initial = minimumBoundingRectangle(parent.filterContoursOutput());
         ArrayList<RotatedRect> findContours = minimumBoundingRectangle(parent.findContoursOutput());
         ArrayList<RotatedRect> ok = new GibberishRectangleFilter().filter(initial);
-        ArrayList<RotatedRect> selected = new DumbAndAloneRectangleFilter().filter(ok);
-
+        numRects = ok.size();
```

Comment:
i would really like to see this logic ( index, numRects, prevNumRects, etc in a separate class. its a lot of stuff cluttering up this class

Reply:
lines 145-147 are at a pretty high level of abstraction: filter some stuff.

but then with 148, we dive into detail logic around determining if we are still locked.